c/tskit/tables.c

```diff
@@ -9206,36 +9385,56 @@ tsk_table_collection_subset(
     }
 
     // mutations and sites
-    i = 0;
+    // make a first pass through to build the mutation_map so that
+    // mutation parent can be remapped even if the table is not in order
```

Comment:
This next bit got more complicated, but `subset` is as a result a lot more robust, not requiring any sort of sorting whatsoever.